ovs/extensions/healthcheck/utils/extension.py

```diff
@@ -0,0 +1,289 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Utilities module for OVS health check
+"""
+
+import subprocess
+import xmltodict
+import commands
+import json
+from ovs.extensions.services.service import ServiceManager
+from ovs.extensions.generic.sshclient import SSHClient
+from ovs.extensions.generic.system import System
+
+
+class _Colors:
+    """
+    Colors for Open vStorage healthcheck logging
+    """
+
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    SKIP = '\033[95m'
+    ENDC = '\033[0m'
+
+    def __init__(self):
+        """ Init method """
+
+        pass
+
+
+class Utils:
+    """
+    General utilities for Open vStorage healthcheck
+    """
+
+    def __init__(self):
+        """ Init method """
+
+        # module specific
+        self.module = "utils"
+
+        # load config file
+        self.settings_loc = "/opt/OpenvStorage/config/healthcheck/settings.json"
+        with open(self.settings_loc) as settings_file:
+            self.settings = json.load(settings_file)
+
+        # fetch from config file
+        self.debug = self.settings["healthcheck"]["debug_mode"]
+        self.max_log_size = self.settings["healthcheck"]["max_check_log_size"]  # in MB
+
+        # open ovs ssh client
+        self.client = SSHClient('127.0.0.1', username='root')
+
+        # init at runtime
+        self.etcd = self.check_etcd()
+        self.serviceManager = self.get_service_manager()
+        self.node_type = self.get_ovs_type()
+        self.ovs_version = self.get_ovs_version()
+        self.cluster_id = self.get_cluster_id()
+
+    def get_config_file_path(self, name, node_id, product, guid=None):
+        """
+        Gets the location of a certain service via local or etcd path
+
+        @param name: name of the PRODUCT (e.g. vpool01 or backend01-abm)
+        @param node_id: the ID of the local node
+        @param product: the id of the desired product
+            * arakoon = 0
+            * vpool = 1
+            * alba_backend = 2
+            * alba_asd = 3
+            * ovs framework = 4
+        @param guid: guid of a certain vpool (only required if one desires the config of a vpool)
+
+        @type name: str
+        @type node_id: str
+        @type product: int
+        @type guid: str
+
+        @return: location of a config file
+
+        @rtype: str
+        """
+
+        # INFO
+        # guid is only for volumedriver (vpool) config and proxy configs
+
+        # fetch config file through etcd or local
+
+        # product_name:
+        #
+        # arakoon = 0
+        # vpool = 1
+        # alba_backends = 2
+        # alba_asds = 3
+        # ovs = 4
+
+        if not self.etcd:
+            if product == 0:
+                return "/opt/OpenvStorage/config/arakoon/{0}/{0}.cfg".format(name)
```

Comment:
Master only uses Etcd. Since the healthcheck is a part of OVS itself, it doesn't make much sense to support classic configuration files.